src/mscorlib/src/System/String.cs

```diff
@@ -666,6 +666,37 @@ private unsafe String CtorCharPtrStartLength(char* ptr, int startIndex, int leng
         [MethodImplAttribute(MethodImplOptions.InternalCall)]
         public extern String(char c, int count);
 
+        private unsafe string CtorCharRefLength(ref char value, int length)
+        {
+            Debug.Assert(length >= 0);
+
+            if (length == 0)
+            {
+                return Empty;
+            }
+
+            try
+            {
+                string result = FastAllocateString(length);
+                fixed (char* dest = &result._firstChar, src = &value)
+                {
+                    wstrcpy(dest, src, length);
+                }
+                return result;
+            }
+            catch (NullReferenceException)
```

Comment:
I tend to agree with @jkotas here. The exception would be if on a large address aware 32-bit executable, but only if we are more likely to roll around to address 0 prior to some other access violation (guard page?) if `length` overflows.